components/usage/pkg/apiv1/usage_test.go

```diff
@@ -556,3 +556,35 @@ func TestReportGenerator_GenerateUsageReportTable(t *testing.T) {
 		})
 	}
 }
+
+func TestUsageService_ReconcileUsageWithLedger(t *testing.T) {
+	dbconn := dbtest.ConnectForTests(t)
+	from := time.Date(2022, 05, 1, 0, 00, 00, 00, time.UTC)
+	to := time.Date(2022, 05, 1, 1, 00, 00, 00, time.UTC)
+
+	// stopped instances
+	dbtest.CreateWorkspaceInstances(t, dbconn, dbtest.NewWorkspaceInstance(t, db.WorkspaceInstance{
+		StoppingTime: db.NewVarcharTime(from.Add(1 * time.Minute)),
+	}))
+
+	// running instances
+	dbtest.CreateWorkspaceInstances(t, dbconn, dbtest.NewWorkspaceInstance(t, db.WorkspaceInstance{}))
+
+	srv := baseserver.NewForTests(t,
+		baseserver.WithGRPC(baseserver.MustUseRandomLocalAddress(t)),
+	)
+
+	v1.RegisterUsageServiceServer(srv.GRPC(), NewUsageService(dbconn, nil, nil))
+	baseserver.StartServerForTests(t, srv)
+
+	conn, err := grpc.Dial(srv.GRPCAddress(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(t, err)
+
+	client := v1.NewUsageServiceClient(conn)
+
+	_, err = client.ReconcileUsageWithLedger(context.Background(), &v1.ReconcileUsageWithLedgerRequest{
```

Comment:
Doesn't assert the right instances were found because we don't yet update the records in any way. Once we start updating, the test here will be extended to do that.